### PR TITLE
Deep links phase2

### PR DESCRIPTION
**Pull request by tuck1s**
Improvements to the Deep Links AND HTTPS documentation, to cover
- Android (new app written and tested in Kotlin, descriptions on autoverify etc)
- Apache (new rules to enable ET forwarding + deep link spec files on same domain)
- Fastly CDN (reworked HTTPS doc, new section in Deep Links doc). Fastly is actually pretty good.

### CODE REVIEW

File: support/tech-resources/using-proxy-https-tracking-domain.md [hunk 1]
```diff
@@ -23,9 +23,13 @@ This post covers how to configure a SparkPost tracking domain, provision an SSL
 
 This example uses a t2.micro Ubuntu instance on Amazon Web Services and a tracking domain of click.nddurant.com that is CNAME'd to the instance public IPv4 DNS.
```
Comment (ndurant-sp) [inline support/tech-resources/using-proxy-https-tracking-domain.md:24]: is this still true in your example for apache?

**Review by ndurant-sp — APPROVED** [review]

File: support/tech-resources/using-proxy-https-tracking-domain.md (hunk 1 above)
Comment (tuck1s) [reply support/tech-resources/using-proxy-https-tracking-domain.md:24]: I worked on my own domain, but left your example in the doc.

File: support/tech-resources/deep-links-self-serve.md [hunk 2]
```diff
@@ -74,30 +74,80 @@ Configure the `paths` section to match the links in your email HTML content, dep
...
-        "package_name": "com.example.deeplinking",
-        "sha256_cert_fingerprints":
-        [<Your_APP_FINGERPRINT>]
-    }
+  "relation": ["delegate_permission/common.handle_all_urls"],
+  "target": {
+    "namespace": "android_app",
+    "package_name": "com.example.testlinks",
+    "sha256_cert_fingerprints":
+    ["xyzzy"]
```
[the remark this answers is not in the document]
Comment (tuck1s) [reply support/tech-resources/deep-links-self-serve.md:82]: Agreed. Will change it.

File: support/tech-resources/enabling-https-engagement-tracking-on-sparkpost.md [hunk 3]
```diff
@@ -267,24 +286,87 @@ If you use AWS CloudFront to enable HTTPS engagement tracking, by default, Cloud
...
+
+![](media/enabling-https-engagement-tracking-on-sparkpost/fastly-stats.png)
+
+
 ---
 
-## <a name = "switch-to-secure"></a> Switch tracking domain to secure, and validate
+## <a name="ttl"></a> Cache Time To Live (TTL) settings
+
+CDNs apply caching, with a "Time to Live" (TTL) for each unique request URL. When a request is first fetched, it is cached. Within the TTL, later requests *may* be served to the client from the CDN cache, without touching the SparkPost endpoint. The client is redirected to the landing page as usual, but a side-effect is that SparkPost does not record the repeat opens/clicks.
```
Comment (rdawemsys) [inline support/tech-resources/enabling-https-engagement-tracking-on-sparkpost.md:360]: @ndurant-sp @ischer For opens, we return `Cache-Control: no-cache, max-age=0`. For clicks, we do not return a Cache-Control header at all. This appears to be true for both Momentum and the new ET service.

Given what Steve found during the preparation of these changes, I think we should consider returning some kind of TTL in a Cache-Control for clicks too.

Both Cloudflare and AWS say they will cache 302 redirects:

 * https://support.cloudflare.com/hc/en-us/articles/200172516-Are-301-and-302-redirects-cached-by-CloudFlare-
 * https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Expiration.html

Interestingly, the now-obsolete old HTTP standard recommended against caching 302s (SHOULD) but the new RFC does not. See old one https://tools.ietf.org/html/rfc2616#section-10.3.3 and new one https://tools.ietf.org/html/rfc7231#section-6.4.3

**Review by rdawemsys — APPROVED** [review]

**Pull request merged**